aviary/subsystems/propulsion/gearbox/gearbox_builder.py

```diff
@@ -0,0 +1,89 @@
+from aviary.subsystems.subsystem_builder_base import SubsystemBuilderBase
+from aviary.subsystems.propulsion.gearbox.model.gearbox_premission import GearboxPreMission
+from aviary.subsystems.propulsion.gearbox.model.gearbox_mission import GearboxMission
+from aviary.variable_info.variables import Aircraft, Dynamic
+
+
+class GearboxBuilder(SubsystemBuilderBase):
+    '''
+    Define the builder for a single gearbox subsystem that provides methods to define the gearbox subsystem's states, design variables, fixed values, initial guesses, and mass names.
+
+    It also provides methods to build OpenMDAO systems for the pre-mission and mission computations of the subsystem, to get the constraints for the subsystem, and to preprocess the inputs for the subsystem.
+
+    This is meant to be computations for a single gearbox, so there is no notion of "num_gearboxs" in this code.
+    '''
+
+    def __init__(self, name='gearbox', include_constraints=True):
+        '''Initializes the GearboxBuilder object with a given name.'''
+        self.include_constraints = include_constraints
+        super().__init__(name)
+
+    def build_pre_mission(self, aviary_inputs):
+        '''Builds an OpenMDAO system for the pre-mission computations of the subsystem.'''
+        return GearboxPreMission(aviary_inputs=aviary_inputs)
+
+    def build_mission(self, num_nodes, aviary_inputs):
+        '''Builds an OpenMDAO system for the mission computations of the subsystem.'''
+        return GearboxMission(num_nodes=num_nodes, aviary_inputs=aviary_inputs)
+
+    def get_design_vars(self):
+        '''
+        Design vars are only tested to see if they exist in pre_mission
+        Returns a dictionary of design variables for the gearbox subsystem, where the keys are the 
+        names of the design variables, and the values are dictionaries that contain the units for 
+        the design variable, the lower and upper bounds for the design variable, and any 
+        additional keyword arguments required by OpenMDAO for the design variable.
+
+                        '''
+
+        DVs = {
+            Aircraft.Engine.Gearbox.GEAR_RATIO: {
+                'opt': True,
+                'units': None,
+                'lower': 1.0,
+                'upper': 20.0,
+                'val':  10  # initial value
+            },
+            Aircraft.Engine.Gearbox.SPECIFIC_TORQUE: {
```

Comment:
Should the upper and lower bounds be the same?

Reply:
No, the goal here was to allow the optimizer to choose the ideal ratio. I guess to keep things simple we could choose a single ratio.

Reply:
I mean for specific_torque.  Lower and Upper are both 100, so there is no reason to even give this dv to the optimizer.

Reply:
I'm going to take out specific_torque for right now and we'll discuss a re-write of the DV vs. parameters in another PR.